### ISSUE TITLE
brmsfit_multiple model comparison

### ISSUE THREAD

**Issue by al-obrien** [paul-buerkner/brms#1002]
When working with a dataset containing explanatory variables with missing data imputed via MICE, a warning message occurs on the `brmsfit_multiple` objects. This was noted as a [false positive](https://discourse.mc-stan.org/t/brm-multiple-warning-with-fitted/10679/2) in some instances, such as when running `fitted()` on new data. However, this may be intended when running [`pp_check`](https://discourse.mc-stan.org/t/how-to-interpret-brm-multiple-warning-when-using-pp-check-and-loo/9189).

I have noticed that this warning also appears when running `add_criterion()` and was hoping if *any additional documentation on how to interpret/heed this warning is available*. Are there no standard approaches available yet for model comparison on data imputed using MICE; is a 'more principled approach' under active development for `brms`? I struggled to find much online. This is important to assess the value of including certain predictors in the model.

Building upon a prior reprex...
```r
library(brms)
library(mice)
data(iris)

# Add missing
iris$Petal.Length[sample(1:nrow(iris), size = 0.1*nrow(iris), replace = F)] <- NA
iris_imp <- mice(iris, m=3, print=F)

mod <- brm_multiple(Sepal.Length~Petal.Length + (1|Species), data = iris_imp, chains=4, iter=2000)
add_criterion(mod, 'waic')
```

The error output is:
`Warning messages:
1: Using only the first imputed data set. Please interpret the results with caution until a more principled approach has been implemented. `

**Comment by torkar** [comment]
Hi, I dabbled around with this a while ago and this might help you (however, I don't do a comparison of models using imputation):
https://github.com/torkar/BDA_in_ESE/blob/master/missing_data_example.R
https://arxiv.org/pdf/1904.00661.pdf

But I'd like to say that using multiple imputation is not something I would recommend. I know it sounds strange since I've used it myself, but it was just for trying out the concept :)

In short, I believe you should model the missingness in your model in a principled Bayesian way, and not impute values. See e.g. 2nd edition of _Statistical Rethinking_ by McElreath (Ch. 15) where he, in addition, covers missingness in binary data.

**Comment by al-obrien** [comment]
Unfortunately I only have the first edition and that only discussed Bayesian imputation of continuous variables. 

I thought MICE was a popular method to impute data, are there some limitations you have come across that would suggest avoiding it? 

I know `brms` supports imputation both prior and during model fitting (from the vignette), but there are challenges in the latter case when the data is discrete and using STAN as the engine. *My question is around if model comparison with `brms` and MICE is not recommended (based on the warning) and if, like you said, a different approach is more accepted.*

**Comment by paul-buerkner** [comment]
brms post-processing of `brm_multiple` is not yet fully developed and only the first data set is stored in the model object (instead of all of them) so that `brm_multiple` models can be made to behave as the usual brm models. See also https://github.com/paul-buerkner/brms/issues/997
